core/SiteContentDetector.php

```diff
@@ -63,56 +56,95 @@ class SiteContentDetector
     /** @var Lazy */
     private $cache;
 
-    /**
-     * @var GtmSiteTypeGuesser
-     */
-    private $siteGuesser;
-
     public function __construct(?Lazy $cache = null)
     {
         if ($cache === null) {
             $this->cache = Cache::getLazyCache();
         } else {
             $this->cache = $cache;
         }
-        $this->siteGuesser = new GtmSiteTypeGuesser();
     }
 
+
     /**
-     * Reset the detection properties
+     * @return array<string, SiteContentDetectionAbstract[]>
+     */
+    public static function getSiteContentDetectionsByType()
```

Comment:
Is there a reason this is static? Also might be good to add a return type while you are here

Reply:
Hm. It's a stateless method, that doesn't work with any instance attributes. So there is actually no reason for having it non static. Could change that, but that would also mean changing `getKnownConsentManagers` to non static as it uses that method. What would be your argument for having it non static?

Reply:
My only reason for non-static is in it's current state it can't be mocked making testing difficult, no strong opinion, up to you.